### DISCUSSION TITLE
postMessage and initBackend

### DISCUSSION THREAD

**Discussion by clockback** [jlongster/absurd-sql#39] [Q&A] [answered]
I'm very keen to use absurd-sql for a web application I'm working on, but I'm quite confused by one issue, and it could be my inexperience talking, so my apologies in advance.

I note that `initBackend` is advised for creating Workers running absurd-sql, to avoid nested Workers on Safari. If I want to pass data back from my Worker using `postMethod`, where does this end up? If I try defining `onmessage` for my Worker, it gives me `MessageEventPrototype` instances them where `event.data.type` is `__absurd:spawn-idb-worker`.

How do I pass data from the Worker as I normally would using `postMethod`?

**Comment by quolpr** [comment]
Here is how I made it:

Worker receive & send message — https://github.com/quolpr/harika/blob/d024b318b012b24a0069e46fe8af74bd191c9a61/packages/web-core/src/extensions/DbExtension/DbWorker.ts#L218

Post & receive messages from worker — https://github.com/quolpr/harika/blob/d024b318b012b24a0069e46fe8af74bd191c9a61/packages/web-core/src/extensions/DbExtension/DB.ts#L128-L141

**Reply by dpmccabe** [reply]
Wow, both the DB and Syncing extensions you've written here would be incredibly valuable extracted into their own packages. Do you have any plans to do so?

**Reply by quolpr** [reply]
I would be happy to do so, and I am making the architecture in such way so it would be easy to decouple, but right now I don't have enough time. Also I am build the version 2, with better sync and sql support

**Comment by quolpr** [comment, marked as answer]
TLDR: just ignore message with `__absurd:spawn-idb-worker.`. Use postMessage/onMessage as usual

**Reply by clockback** [reply]
Ah, thanks. I don't think I'd thought about it properly, but in my first uses, I only needed to receive a single message, so I terminated after the first, thus I didn't realize that the plain message was also sent.
